lib/streamlit/watcher/LocalSourcesWatcher.py

```diff
@@ -56,6 +53,20 @@
             % msg
         )
 
+watcherType = config.get_option("server.fileWatcherType")
+
+if watcherType == "auto":
+    if watchdog_available:
+        FileWatcher = EventBasedFileWatcher
+    else:
+        from streamlit.watcher.PollingFileWatcher import PollingFileWatcher
+        FileWatcher = PollingFileWatcher
+elif watcherType == "watchdog" and watchdog_available:
+    FileWatcher = EventBasedFileWatcher
+elif watcherType == "poll":
+    from streamlit.watcher.PollingFileWatcher import PollingFileWatcher
+    FileWatcher = PollingFileWatcher
```

Comment:
I was purposely trying to throw out the exception, it would be captured and then return the same way. As such, in case it's mistakenly set as None, (I don't think this should be used, at least, not most of the cases), at least they can get an alert.

Happy to skip throwing & capturing the exception if you insist.

Reply:
> Actually, looking at the test you added, I think this whole code block should be pulled in to a separate function. This way we can test is really easily without the need to call `reload` every time.
> 
> Like this:
> 
> ```python
> def get_file_watcher_class():
>     watcherType = config.get_option("server.fileWatcherType")
> 
>     if watcherType == "auto":
>          if watchdog_available:
>             return EventBasedFileWatcher
>         else:
>             from streamlit.watcher.PollingFileWatcher import PollingFileWatcher
>             return PollingFileWatcher
>     elif watcherType == "watchdog" and watchdog_available:
>         return EventBasedFileWatcher
>     elif watcherType == "poll":
>         from streamlit.watcher.PollingFileWatcher import PollingFileWatcher
>         return PollingFileWatcher
>     else:
>         return None
> 
> FileWatcher = get_file_watcher_class()
> ```
> 
> ...and then in the tests you just need to test `get_file_watcher_class()`

Updated, thanks.